slam_toolbox/solvers/ceres_solver.cpp

```diff
@@ -37,7 +38,9 @@ CeresSolver::CeresSolver() :
   first_node_ = nodes_->end();
 
   // formulate problem
-  angle_local_parameterization_ = AngleLocalParameterization::Create();
+  local_parameterization_ = new ceres::ProductParameterization(
```

Comment:
This is `new`ed but never deleted in reset /  destructor

Reply:
Ahh, good catch. Fixed in https://github.com/SteveMacenski/slam_toolbox/pull/468/commits/cb7de4112e5798741e9882a439e9e956334b9695.

Reply:
BTW what's the point of dynamically allocating most `CeresSolver` member fields? Barring those that must be because Ceres' API requires it, it doesn't seem to add much value -- it only opens the door to leakage if constructor, destructor, and reset implementations ever end up out of sync.

Reply:
I think the things that are dynamically allocated are thoes that Ceres requires.

Reply:
Hmm, those `std::unordered_map` members are not required by Ceres.

Reply:
I don't understand what you're referencing this comment is on `AngleLocalParameterization` which is a ceres-owned object . I personally don't like that I had it raw pointers and the expectation is that it deals with the lifecycle, but that's what Ceres docs said when I was looking at building this

Reply:
> I don't understand what you're referencing this comment is on `AngleLocalParameterization`

Yes, you're right, my bad. I was referring to e.g. 
https://github.com/SteveMacenski/slam_toolbox/blob/646ef41b14e2b5f72e721963fafa490afa3d64fc/solvers/ceres_solver.hpp#L72-L73
Anyhow, it's orthogonal to this patch.